### DISCUSSION TITLE
What's coolest about next.js in your opinion?

### DISCUSSION THREAD

**Discussion by sebastianjnuwu** [vercel/next.js#45498] [Help] [answered]
I want the opinion of those who use it, I'm seeing if I change ejs for next.js....

**Comment by guitheengineer** [comment, marked as answer]
I've been working with nextjs for 2 years now to build medium/large scale apps and in my experience It's the way it abstracts great powerful features (like incremental static generation) to use in such an easy manner. Trying to build a feature like that from scratch would cost literally thousands of $$$ due to paid hours going for architecturing, developing and testing it. The way I see nextjs is as a great collection of tools to build performant applications with good DX (considering next/image, next/head, gSP, gSSP, next/scripts, file system routing, middlewares, etc)
